runtime/src/system_instruction_processor.rs

```diff
@@ -361,28 +366,44 @@ pub fn process_instruction(
                 );
                 return Err(NonceError::NoRecentBlockhashes.into());
             }
-            me.advance_nonce_account(&signers, invoke_context)
+            advance_nonce_account(
+                invoke_context,
+                instruction_context,
+                &signers,
+                NONCE_ACCOUNT_INDEX,
```

Comment:
Why do we pass this in, looks like it's always `NONCE_ACCOUNT_INDEX`

Reply:
This was just to get the tests to stay compatible with those in nonce_keyed_account.
It was overthrown in #23217 and replaced by the `instruction_account_indices` enums.